### ISSUE TITLE
node package jquery uses window.setTimeout 

### ISSUE THREAD

**Issue by ArndBrugman** [jquery/jquery#2906]
I use jquery in node for my hue-api called huepi combined with domino and xmlhttprequest. see https://github.com/arndbrugman/huepi 

Untill recently this all worked fine. Latest release from npm gives an error that window.setTimeout is not defined. (jquery.js:9157:15), as window is not defined.

Example:
setTimeout(function(){console.log('done')}, 1200); // works in node
window.setTimeout(function(){console.log('done')}, 1200); // doesn't work because window is undefined

could you please use setTimeout without window in the node release.

Thx.

**Comment by ArndBrugman** [comment]
Found a total of 3 window.setTimeout at:

src/queue/delay.js line 14
src/core/ready.js line 79
and probably the one that is causing my code not to be executed:
src/ajax/xhr.js line 133

Thx.

**Comment by dmethvin** [comment]
There is no "node release", the code posted on npm runs in all supported environments. What exactly is the environment you are running on? Is it using something like jsdom

See the discussion for this release:
http://blog.jquery.com/2016/01/08/jquery-2-2-and-1-12-released/ (search "setTimeout")
https://github.com/jquery/jquery/issues/2177

**Comment by mgol** [comment]
Thanks for the report.

I see you use the [domino](https://github.com/fgnass/domino) module to provide a `window` implementation on the Node.js side. This module doesn't define `window.setTimeout` etc. which is why it breaks for you.

jQuery to work outside of a browser requires a `window` implementation that has all the browser APIs that jQuery uses. We've switched to use `window.setTimeout` because taking almost all browser globals from `window` and only 4 of them (`setTimeout`, `clearTimeout`, `setInterval`, `clearInterval`) from the global was inconsistent and meant if someone run jQuery in an environment that doesn't have those globals but provide a complete `window` implementation with all those methods it would still not work. We've had bug reports for that, see e.g. #2177. To fix those issues we've considered using `window.setTimeout` and fall back to the global one when not provided in PR #2391 but it would just add to size & complexity so we decided to just be consistent and take it all from the provided `window`.

You have two choices:

1. Switch to a `window` implementation that provides `setTimeout` & friends.
2. Submit a PR to [domino](https://github.com/fgnass/domino) adding support for those APIs.
